aspnetcore/blazor/security/server/blazor-web-app-with-oidc.md

```diff
@@ -0,0 +1,378 @@
+---
+title: Secure an ASP.NET Core Blazor Web App with OpenID Connect (OIDC)
+author: guardrex
+description: Learn how to secure a Blazor WebAssembly App with OpenID Connect (OIDC).
+monikerRange: '>= aspnetcore-8.0'
+ms.author: riande
+ms.custom: mvc
+ms.date: 01/26/2024
+uid: blazor/security/server/blazor-web-app-oidc
+zone_pivot_groups: blazor-web-app-oidc-specification
+---
+# Secure an ASP.NET Core Blazor Web App with OpenID Connect (OIDC)
+
+<!-- UPDATE 9.0 Activate after release and INCLUDE is updated
+
+[!INCLUDE[](~/includes/not-latest-version.md)]
+
+-->
+
+This article describes how to secure a Blazor Web App with [OpenID Connect (OIDC)](https://openid.net/developers/how-connect-works/) using a sample app in the [`dotnet/blazor-samples` GitHub repository (.NET 8 or later)](https://github.com/dotnet/blazor-samples).
+
+:::zone pivot="blazor-web-app"
+
+This version of the article covers implementing OIDC without adopting the [Backend for Frontend (BFF) pattern](/azure/architecture/patterns/backends-for-frontends). Change the article version selector to **Blazor Web App (BFF pattern)** if the app's specification calls for adopting the BFF pattern.
+
+The following specification is covered:
+
+* The Blazor Web App adopts [global Auto interactivity](xref:blazor/components/render-modes).
+* Custom auth state provider services are used by the server and client apps to capture the user's authentication state and flow it between the server and client.
+* This app is a starting point for any OIDC authentication flow. OIDC is configured manually in the app and doesn't rely upon [Microsoft Entra ID](https://www.microsoft.com/en-us/security/business/microsoft-entra) or [Microsoft Identity Web](/entra/msal/dotnet/microsoft-identity-web/) packages, nor does the sample app require [Microsoft Azure](https://azure.microsoft.com/) hosting. However, the sample app can used with Entra, Microsoft Identity Web, and hosted in Azure.
```

Comment:
I can create another sample using the Microsoft Identity Web packages if you think that's useful.

Reply:
I'm not sure ... @danroth27 and @mkArtakMSFT will have a better idea of what devs are asking for in this area.

If we add guidance for MS Identity Web/Azure, there are two avenues that we can take ...

* Two articles (might be better from an SEO perspective)
  * We continue with this article on OIDC that has two pivots ... non-BFF pattern and BFF pattern.
  * I create a new sample and ***write a new sister article*** to appear SxS with this article.
* One article with a third pivot
  * We change the name of this article to something more generic like *Blazor Web App authentication*.
  * I add the sample app, and I ***add a pivot to this article to cover it***, so the pivots would be "OIDC," "OIDC (BFF pattern)," and "MS Identity Web with Azure hosting" ... something like that.

Either way, I'd plan to go ahead and finish this up and get it live tomorrow (Tuesday). I'll create a new issue/PR for the work. I'd like to get this *LIVE*⚡*ASAP* 🏃.

Reply:
@guardrex @halter73 

I have an Entra example here:

https://github.com/damienbod/Hostedblazor8MeID/tree/main/BlazorWebMeID

devs need an OIDC solution example and a Microsoft.Identity.Web example for the 4 Entra products. It is recommended to use the Microsoft.Identity.Web packages for Entra due to the OBO support and the CA challenges. OIDC is used for all other IDPs. You can just point them to the https://github.com/AzureAD/microsoft-identity-web/wiki for the docs for Entra maybe?

Reply:
I would use the  **https://github.com/AzureAD/microsoft-identity-web/wiki**  instead of **/entra/msal/dotnet/microsoft-identity-web/**

Reply:
Thanks @damienbod ... I'm 👂 for @danroth27 / @mkArtakMSFT to get back to us on *IF* we'll provide content on it and *HOW* we'll do it ☝️. If so, then I think @halter73 will decide if we can take your solution as the example.

WRT the link, that's a great additional resource ... I'll add it, but I must link to the public site's product page on first mention ... ***without the `en-us` segment!** 🙈😆 ... I'll 🔪 that segment out of there in a sec.

Reply:
We have an answer on this from Jeremy offline. There's a plan on the Azure docs side to cover BWA with MS Identity Web/Azure hosting. Therefore, we don't need to sweat it over here. I'll create a tracking reminder for myself that we'll want to cross-link their new content+sample(s) when they appear.